lib/ruby_lsp/event_emitter.rb

```diff
@@ -53,6 +53,12 @@ def emit_for_target(node)
 
     # Visit dispatchers are below. Notice that for nodes that create a new scope (e.g.: classes, modules, method defs)
     # we need both an `on_*` and `after_*` event. This is because some requests must know when we exit the scope
+    sig { override.params(node: T.nilable(SyntaxTree::Node)).void }
+    def visit(node)
+      @listeners[:on_visit]&.each { |l| T.unsafe(l).on_visit(node) }
```

Comment:
Let's call this `on_node` since listeners only deal with node types and have no knowledge of the underlying visitor implementation.

Reply:
Good call, updated 👍